### ISSUE TITLE
Can't enable child blocks of disabled blocks

### ISSUE THREAD

**Issue by gpfernandezflorio** [google/blockly#6719]
**Describe the bug**

When a block is disabled, all its children turn disable too, which makes sense. The problem is that you can't re-enable any of the child blocks. The option in the dropdown menu appears disabled.

I guess the problem is that the field 'diabled' does not update in the child block.

**To Reproduce**

1. Nest one block inside another. For example, one if inside another if.
2. Disable the outer if with the dropdown menu option.
3. Right click on the inner if to open its dropdown menu.
4. The option is diabled.

**Expected behavior**

I expect the option "Enable block" to appear normally in the dropdown menu of the inner block, just as it appears in the outer one.

**Screenshots**

![image](https://user-images.githubusercontent.com/5846063/208898298-0c2a8572-b36a-4eea-853f-f30d3309fee9.png)
![image](https://user-images.githubusercontent.com/5846063/208898347-33b7f77f-2250-4ea5-968d-e0bc67889882.png)
![image](https://user-images.githubusercontent.com/5846063/208898618-3e3c6121-1fee-4932-84fe-2ad8fca9ad1b.png)

**Additional context**

I ran it on the web.

**Comment by gonfunko** [comment]
Thanks for raising this issue! I believe this is working as intended, since it's somewhat conceptually unclear how this would work if e.g. a loop block was disabled but its child blocks were enabled - should they become part of the block stack as if they were just sequential, or become their own independent block stack or something else? Do you have a particular usecase in mind where enabling child blocks of a disabled block would be desirable?

**Comment by gpfernandezflorio** [comment]
I hadn't thought of that.
Actually, my problem was that I needed to retrieve all enabled blocks so I filter all blocks with "disabled" field in false. That's how I ended discovering that child blocks are visually disabled but have the "disabled" field in false.
Maybe you're right and what I reported is the intended behaviour but maybe it would be better if the disabled option was "Enable block" instead of "Disable block" and the field "disabled" gets the value true.
Or maybe there is another way of retrieving disabled blocks instead of reading that field.

**Comment by albertg361** [comment]
I think enable the child bloks and merge then

**Comment by gpfernandezflorio** [comment]
What do you mean with "merge"? Merge with what?

**Comment by gpfernandezflorio** [comment]
Ok, I found there is a function isEnabled and another function getInheritedDisabled. I can solve my original problem with them so I'm closing this issue.

I still think the text in the disable option of the dropdown menu should be "Enable block" instead of "Disable block" to avoid confusion. That can be done changing the displayText field of registerDisable to:
`      return scope.block!.isEnabled() &&
        !scope.block!.getInheritedDisabled() ? Msg['DISABLE_BLOCK'] :
                                               Msg['ENABLE_BLOCK'];`

This is what I got changing only that:
![image](https://user-images.githubusercontent.com/5846063/209684692-b287deda-0263-4b95-8ed7-5dd196396131.png)

I also suggest changing isEnabled to return False if the block is visually disabled for being a child of a disabled block (it currently returns `!this.disabled;`) but I understand that this change may break something else.
